app/com/linkedin/drelephant/exceptions/core/ExceptionFingerprintingSpark.java

```diff
@@ -0,0 +1,368 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.drelephant.exceptions.core;
+
+import com.linkedin.drelephant.ElephantContext;
+import com.linkedin.drelephant.analysis.AnalyticJob;
+import com.linkedin.drelephant.exceptions.Classifier;
+import com.linkedin.drelephant.exceptions.ExceptionFingerprinting;
+import com.linkedin.drelephant.exceptions.util.ExceptionInfo;
+import com.linkedin.drelephant.spark.fetchers.statusapiv1.StageData;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import models.JobDefinition;
+import models.JobExecution;
+import models.TuningJobDefinition;
+import org.apache.log4j.Logger;
+import org.apache.hadoop.conf.Configuration;
+
+import static com.linkedin.drelephant.exceptions.util.ExceptionInfo.*;
+import static com.linkedin.drelephant.exceptions.util.Constant.*;
+import static com.linkedin.drelephant.exceptions.util.ExceptionUtils.*;
+import static com.linkedin.drelephant.exceptions.util.ExceptionUtils.ConfigurationBuilder.*;
+
+
+public class ExceptionFingerprintingSpark implements ExceptionFingerprinting {
+
+  private static final Logger logger = Logger.getLogger(ExceptionFingerprintingSpark.class);
+  private boolean debugEnabled = logger.isDebugEnabled();
+  private static final String JOBHISTORY_WEBAPP_ADDRESS = "mapreduce.jobhistory.webapp.address";
+  private static final String NODEMANAGER_ADDRESS = "yarn.nodemanager.address";
+  private static final int STARTING_INDEX_FOR_URL = 2;
+  private static final String CONTAINER_PATTERN = "container";
+  private static final String HOSTNAME_PATTERN = ":";
+  private static final String URL_PATTERN = "/";
+  private static final String JOBHISTORY_ADDRESS_FOR_LOG = "http://{0}/jobhistory/nmlogs/{1}";
+
+  private enum LogLengthSchema {LOG_LENGTH, LENGTH}
+
+  private AnalyticJob analyticJob;
+  private List<StageData> failedStageData;
+  private boolean useRestAPI = true;
+
+  static {
+    ConfigurationBuilder.buildConfigurations(ElephantContext.instance().getAutoTuningConf());
```

Comment:
Would it be good to keep EF configuration in different configuration file. As our long term goal is to classify broader exception.

Reply:
There should be JobType specific configuration as well in this. Can be handled when we add exception fingerprinting for MapReduce job.

Reply:
Yes , it make sense to have separate configuration file for Exception Fingerprinting. In the current scope , its for autotuning hence taking from AutoTuningConf. I'll move it out as separate PR